src/utils.c

```diff
@@ -49,14 +49,34 @@ int cat_file(char *path) {
 
 /* Read a tracefs file to stdout */
 int cat_tracefs_file(char *tracefs, char *fn) {
-	char tracef[100];
-
-	tracef[0] = 0;
-	strcat(tracef, tracefs);
-	strcat(tracef, "/");
-	strcat(tracef, fn);
-
-	return cat_file(tracef);
+/*	char tracef[100];*/
+
+/*	tracef[0] = 0;*/
+/*	strcat(tracef, tracefs);*/
+/*	strcat(tracef, "/");*/
+/*	strcat(tracef, fn);*/
+
+	char * tracef;
+	size_t len_tfs, len_fn;
+	int    return_val;
+	
+	len_tfs = strlen(tracefs);
+	len_fn  = strlen(fn);
+	
+	/* 1 for the '\0', and one for the '/' */
+	tracef = malloc(len_tfs+len_fn+2);
+	
+	/* I opted to use memcpy instead of strcat because I only wanted to
+	itterate each string once. */
+	memcpy(tracef, tracefs, len_tfs);
+	memcpy(tracef+len_tfs+1, fn, len_fn);
+	
+	tracef[len_tfs] = '/';
+	tracef[len_tfs+len_fn+1] = '\0';
+	
+	return_val = cat_file(tracef);
```

Comment:
thanks for the effort, could you follow the style guidelines before posting:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html